utils/coreError.ts

```diff
@@ -0,0 +1,19 @@
+class CoreError extends Error {
+  code?: string;
+
+  constructor(message: string, code?: string, causedBy?: Error) {
+    super(message);
+    this.name = 'CoreError';
+    this.code = code;
+
+    if (causedBy && causedBy.stack) {
+      this.stack += `\n\nCaused by: \n${causedBy.stack}`;
+    }
+  }
+
+  static isCoreError(e: Error): e is CoreError {
+    return e.name === 'CoreError';
+  }
+}
+
+export { CoreError };
```

Comment:
can we export this?

Reply:
Yes, that is my bad 😅 Thanks for catching